src/webui/www/public/upload.html

```diff
@@ -6,69 +6,61 @@
-		xhr.send(formData);
-	}
-}
-
-function fileHandler(e) {
-	e.preventDefault();
-	$('upload_spinner').style.display = "block";
-	// fetch FileList object
-	var files = $('fileselect').files
-	// process all File objects
-	uploadFiles(files);
-}
-
-window.addEvent('load', function() {
-	// is XHR2 available?
-	var xhr = new XMLHttpRequest();
-	if (xhr.upload) {
-		$('uploadForm').addEvent('submit', fileHandler);
-	} else {
-		$('upload_frame').addEvent('load', function() { window.parent.closeWindows(); });
-		$('uploadForm').addEvent('submit', function() { $('upload_spinner').style.display = "block"; });
-	}
-});
-</script>
 </head>
 <body>
 <iframe id="upload_frame" name="upload_frame" class="invisible" src="javascript:false;"></iframe>
-<form action="command/upload" enctype="multipart/form-data" method="post" id="uploadForm" target="upload_frame" style="text-align: center;">
-<div style="margin-top: 25px; display: inline-block; border: 1px solid lightgrey; border-radius: 4px;">
-	<input type="file" id="fileselect" name="fileselect[]" multiple="multiple" />
+<form action="command/upload" enctype="multipart/form-data" method="post" id="uploadForm" style="text-align: center;" target="upload_frame">
+<p>
+	<div style="margin-top: 25px; display: inline-block; border: 1px solid lightgrey; border-radius: 4px;">
+		<input type="file" id="fileselect" name="fileselect[]" multiple="multiple"/>
+	</div>
+</p>
+<fieldset class="settings" style="border: 0; text-align: left;">
+<div class="formRow" style="margin-top: 12px;">
+	<label for="savepath" class="leftLabelLarge">QBT_TR(Save files to location:)QBT_TR</label>
+	<input type="text" id="savepath" name="savepath" style="width: 16em;"/>
+</div>
+<div class="formRow">
+	<label for="label" class="leftLabelLarge">QBT_TR(Label:)QBT_TR</label>
+	<input type="text" id="label" name="label"/ style="width: 16em;"/>
 </div>
-<div id="submitbutton" style="margin-top: 30px;">
+<div id="submitbutton" style="margin-top: 30px; text-align: center;">
 	<button type="submit" style="font-size: 1em;">QBT_TR(Upload Torrents)QBT_TR</button>
 </div>
+</fieldset>
 </form>
+<script type="text/javascript">
+
+var submitted = false;
+
+$('uploadForm').addEventListener("submit", function() {
+	$('upload_spinner').style.display = "block";
+	submitted = true;
+});
+
+$('upload_frame').addEventListener("load", function() {
+	if (submitted)
+		window.parent.closeWindows();
+});
+
+function getSavePath() {
+        var req = new Request({
+                url: 'command/getSavePath',
+                method: 'get',
+                noCache: true,
+                onFailure: function() {
+                        alert("Could not contact qBittorrent");
+                },
+                onSuccess: function(data) {
+                        if (data) {
+                                  $('savepath').setProperty('value', data);
+                        }
+                }
+        }).send();
+}
+
+getSavePath();
```

Comment:
I can do it like this:

```javascript
$(window).addEventListener("load", function() {
        getSavePath();
});
```

We can't use the jQuery's ready() function since we're not using that framework.  It has to be "pure javascript".